### PR TITLE
Nullspace vecnorm merge

### PR DESCRIPTION
**Pull request by petropusz**
PR for merged code that has both Jarek's nullspace and clustering & quantization scripts updated with normalization option. This uses code from repos & branches https://github.com/facebookresearch/CPC_audio/tree/zerospeech, https://github.com/tuanh208/CPC_audio/tree/zerospeech, https://github.com/bootphon/zerospeech2021_baseline in addition to facebookresearch/CPC_audio master branch.

### CODE REVIEW

File: cpc/cpc_default_config.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def set_default_cpc_config(parser):
                        choices=['reverse', 'none'],
                        help='Some variations on CPC.')
     group.add_argument('--encoder_type', type=str,
-                       choices=['cpc', 'mfcc', 'lfb'],
+                       choices=['cpc', 'mfcc', 'lfb', 'smart'],
```
Comment (jdzikowski) [inline cpc/cpc_default_config.py]: Remove this, this is for smartpooling which we do not use in this work

File: cpc/feature_loader.py [hunk 2]
```diff
@@ -138,6 +142,9 @@ def getEncoder(args):
     elif args.encoder_type == 'lfb':
         from .model import LFBEnconder
         return LFBEnconder(args.hiddenEncoder)
+    elif args.encoder_type == 'smart':
```
Comment (jdzikowski) [inline cpc/feature_loader.py]: Same thing, remove this block. I made a mistake adding this.

File: cpc/model.py [hunk 3]
```diff
@@ -104,6 +104,206 @@ def forward(self, x):
         x = F.relu(self.batchNorm4(self.conv4(x)))
         return x
 
+class Smartpool(nn.Module):
```
Comment (jdzikowski) [inline cpc/model.py]: Remove Smartpool, DoXTimes and CPCSmartpool encoder

**Review by jdzikowski — CHANGES_REQUESTED** [review]
Remove the following changes - they're meant for smartpooling which we didn't use in zerospeech at all and it's experimental

File: cpc/cpc_default_config.py (hunk 1 above)
Comment (petropusz) [reply cpc/cpc_default_config.py]: done

File: cpc/feature_loader.py (hunk 2 above)
Comment (petropusz) [reply cpc/feature_loader.py]: done

File: cpc/model.py (hunk 3 above)
Comment (petropusz) [reply cpc/model.py]: done

**Comment by petropusz** [comment]
removed smartpooling code
